pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go

```diff
@@ -444,8 +453,17 @@ func (m *managedFolderAlertActionsRepeatMigrator) Exec(sess *xorm.Session, mg *m
 
 	for id, a := range mapped {
 		for scope, p := range a {
+			// previous migration added this permission, but it was not added to the toAdd slice
+			// because we were checking all permissions on top of folders, not just the scoped ones
+			//
+			// what we had:
+			// if !hasAction(ac.<Action>, permissions) {
+			// should have been:
+			// if !hasAction(ac.<Action>, p) {
+			//
+			// see PR for explanation: https://github.com/grafana/grafana/pull/58054
 			if hasFolderView(p) {
-				if !hasAction(ac.ActionAlertingRuleRead, permissions) {
+				if !hasAction(ac.ActionAlertingRuleRead, p) {
```

Comment:
nit: nested conditional statements can be concatenated instead

Reply:
we'll keep the change minimal to point out the change of the migration bug, but it's a good statement!